src/DataSource/Http.hs

```diff
@@ -82,10 +83,11 @@ getTicketInfo ticketId = do
 
     let url = showURL $ TicketsURL ticketId
     let req = apiRequest cfg url
-
-    apiCall <- asksHTTPNetworkLayer hnlApiCall
-
-    Just <$> apiCall parseTicket req
+    case req of
```

Comment:
Chain computation?

Reply:
Means, "use a Monad". You first need to address the other issues.

Reply:
These computations are chained. If they were not, the code wouldn't compile.

Here we have the resulting monad `m` and the `Either` monad. They could be combined in a monad stack, but given that there is a single point where a failure could happen, a monad stack would be overkill. Don't you agree?